src/core/service.c

```diff
@@ -280,24 +281,26 @@ usec_t service_restart_usec(Service *s) {
          * between job enqueuing and running is usually neglectable compared to the time
          * we'll be sleeping. */
         n_restarts = s->n_restarts +
-                     (IN_SET(s->state, SERVICE_DEAD, SERVICE_FAILED, SERVICE_AUTO_RESTART) ? 1 : 0);
+                     (IN_SET(s->state, SERVICE_DEAD_BEFORE_AUTO_RESTART, SERVICE_FAILED_BEFORE_AUTO_RESTART, SERVICE_AUTO_RESTART) ? 1 : 0);
```

Comment:
Hmm, if we're doing this, I think we can add 1 to it unconditionally, and change `RestartUSecCurrent` to `RestartUSecNext` instead.

If the service is running, after this change we'll still calculate the interval used last time, but logging it as "Next interval", which is spurious.

Reply:
I wanted this PR to be as minimal as possible, the only thing other than the state engine states I added was to add debug logging here. I am think we definitely cna make further improvements on the timing calculation, but let's keep this out of this PR, to make it easily reviable.

I'll drop the word "next" from the log message for now, to make this less misleading.

Would love to review a follow-up patch adding what you are suggesting!